UPGRADE-2.1.md

```diff
@@ -3,11 +3,23 @@
 ### General
 
 1. The `sylius_admin_customer_orders_statistics` route has been deprecated.
-
 1. The minimum version of Symfony 7 packages has been bumped from Symfony `^7.1` to `^7.2`
-
 1. The `tabler` package has been updated to version `^1.3.0`. Please pay attention to the accordion element in final applications, as its implementation has changed.
 
+## Grid providers are now configurable
```

Comment:
Let's use AI to rewrite this message. (I can do it later, need to go for now)

Key points.
- This is about YAML/PHP and PHP service config formats (which is not clear from current message).
- Sylius 3.0 will use and define PHP service grids only.
- There is no configs merge between YAML/PHP and PHP services formats.
- We providing Migration Layer at `sylius_core` until sylius 3.0, which will allow to force switching format for existing grids, so end-user would be able to switch `sylius_order` from yaml to php service in order to extend it in app level code.

Reply:
Maybe we should also include the link to a possible migration tool.

Reply:
@diimpp Would you be so kind and come up with a better description on the upgrade guide to finish off this PR? The current iteration is the best I could come up with even with AI. :D